networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/network/NetworkConfig.java

```diff
@@ -114,7 +155,71 @@ public boolean isLogWirePlain() {
     return logWirePlain;
   }
 
-  public boolean isLogMuxFrames() {
-    return logMuxFrames;
+  public boolean isLogWireMuxFrames() {
+    return logWireMuxFrames;
+  }
+
+  public boolean isLogWireGossip() {
+    return logWireGossip;
+  }
+
+  public NetworkConfig setGossipD(int gossipD) {
```

Comment:
We've been trying to move more towards objects being immutable so would be good to just inject anything configurable into the constructor rather than having setters.

Reply:
Made all immutable, but also made separate classes to inject defaults in a more convenient way.

Reply:
The extra config classes are a great idea.